staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/conversion_test.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"reflect"
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/apis/example"
+	examplev1 "k8s.io/apiserver/pkg/apis/example/v1"
+	appsv1register "k8s.io/kubernetes/pkg/apis/apps/v1"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+)
+
+func initiateScheme() *runtime.Scheme {
+	s := runtime.NewScheme()
+	example.AddToScheme(s)
+	examplev1.AddToScheme(s)
+	extensions.AddToScheme(s)
+	appsv1register.AddToScheme(s)
+	return s
+}
+
+func TestConvertToGVK(t *testing.T) {
+	scheme := initiateScheme()
+	w := GenericAdmissionWebhook{
+		convertor: scheme,
+		creator:   scheme,
+	}
+	table := []struct {
+		obj         runtime.Object
+		gvk         schema.GroupVersionKind
+		expectedObj runtime.Object
+	}{
+		{
+			obj: &example.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "pod1",
+					Labels: map[string]string{
+						"key": "value",
+					},
+				},
+				Spec: example.PodSpec{
+					RestartPolicy: example.RestartPolicy("never"),
+				},
+			},
+			gvk: examplev1.SchemeGroupVersion.WithKind("Pod"),
+			expectedObj: &examplev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "pod1",
+					Labels: map[string]string{
+						"key": "value",
+					},
+				},
+				Spec: examplev1.PodSpec{
+					RestartPolicy: examplev1.RestartPolicy("never"),
+				},
+			},
+		},
+		// TODO: move the replicaset definition to k8s.io/apiserver
```

Comment:
why do we need another type?

Reply:
I should had removed the todo. It's left there when I used the k8s.io/api's replicaset to prototype the test.

> why do we need another type?

deads2k asked for a test covers cross-group conversion, thus adding the replicaset type to k8s.io/apiserver

Reply:
I named the test so hopefully the intention is obvious now.